exporter/cassandraexporter/exporter_logs.go

```diff
@@ -0,0 +1,133 @@
+
+package cassandraexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/cassandraexporter"
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/gocql/gocql"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/traceutil"
+)
+
+type logsExporter struct {
+	client *gocql.Session
+	logger *zap.Logger
+	cfg    *Config
+}
+
+func newLogsExporter(logger *zap.Logger, cfg *Config) (*logsExporter, error) {
+	cluster := gocql.NewCluster(cfg.DSN)
+	session, err := cluster.CreateSession()
+	cluster.Keyspace = cfg.Keyspace
+	cluster.Consistency = gocql.Quorum
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &logsExporter{logger: logger, client: session, cfg: cfg}, nil
+}
+
+func initializeLogKernel(cfg *Config) error {
+	ctx := context.Background()
+	cluster := gocql.NewCluster(cfg.DSN)
+	cluster.Consistency = gocql.Quorum
+	session, err := cluster.CreateSession()
+	if err != nil {
+		return err
+	}
+
+	createDatabaseError := session.Query(parseCreateDatabaseSQL(cfg)).WithContext(ctx).Exec()
+	if createDatabaseError != nil {
+		return createDatabaseError
+	}
+	createLogTableError := session.Query(parseCreateLogTableSQL(cfg)).WithContext(ctx).Exec()
+	if createLogTableError != nil {
+		return createLogTableError
+	}
+
+	defer session.Close()
+
+	return nil
+}
+
+func (e *logsExporter) Start(ctx context.Context, host component.Host) error {
+	initializeErr := initializeLogKernel(e.cfg)
+	return initializeErr
+}
+
+func (e *logsExporter) Shutdown(_ context.Context) error {
+	if e.client != nil {
+		e.client.Close()
+	}
+
+	return nil
+}
+
+func parseCreateLogTableSQL(cfg *Config) string {
+	return fmt.Sprintf(createLogTableSQL, cfg.Keyspace, cfg.LogsTable, cfg.Compression.Algorithm)
+}
+
+func (e *logsExporter) pushLogsData(ctx context.Context, ld plog.Logs) error {
+	start := time.Now()
+
+	var serviceName string
+	for i := 0; i < ld.ResourceLogs().Len(); i++ {
+		logs := ld.ResourceLogs().At(i)
+		res := logs.Resource()
+		resAttr := attributesToMap(res.Attributes().AsRaw())
+		if v, ok := res.Attributes().Get(conventions.AttributeServiceName); ok {
+			serviceName = v.Str()
+		}
+		for j := 0; j < logs.ScopeLogs().Len(); j++ {
+			rs := logs.ScopeLogs().At(j).LogRecords()
+			for k := 0; k < rs.Len(); k++ {
+				r := rs.At(k)
+				logAttr := attributesToMap(r.Attributes().AsRaw())
+				bodyByte, _ := json.Marshal(r.Body().AsRaw())
+
+				insertLogError := e.client.Query(fmt.Sprintf(insertLogTableSQL, e.cfg.Keyspace, e.cfg.LogsTable),
+					r.Timestamp().AsTime(),
+					traceutil.TraceIDToHexOrEmptyString(r.TraceID()),
+					traceutil.SpanIDToHexOrEmptyString(r.SpanID()),
+					uint32(r.Flags()),
+					r.SeverityText(),
+					int32(r.SeverityNumber()),
+					serviceName,
```

Comment:
I understand that the service name is pretty important. But does it really need a dedicated column?

Do you need it for indexing? I believe you can also make an index for a map key...

Reply:
Why did you resolve it without replying or changing anything? :)